bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/transceiver/EnOceanESP2Transceiver.java

```diff
@@ -128,30 +137,35 @@ protected void processMessage(byte firstByte) {
                                     }
                                 } else {
                                     if (dataBuffer[1] != (byte) 0xFC) {
-                                        logger.debug("Unknown/unsupported ESP2Packet: {}",
-                                                HexUtils.bytesToHex(Arrays.copyOf(dataBuffer, dataLength)));
+                                        byte[] array = Arrays.copyOf(dataBuffer, dataLength);
+                                        String packetString = array != null ? HexUtils.bytesToHex(array) : "";
+                                        logger.debug("Unknown/unsupported ESP2Packet: {}", packetString);
                                     }
                                 }
                             } else {
                                 logger.debug("ESP2Packet malformed: {}", HexUtils.bytesToHex(dataBuffer));
                             }
 
-                            state = _byte == ESP2Packet.ENOCEAN_ESP2_FIRSTSYNC_BYTE
+                            state = byteBuffer == ESP2Packet.ENOCEAN_ESP2_FIRSTSYNC_BYTE
                                     ? ReadingState.WaitingForSecondSyncByte
                                     : ReadingState.WaitingForFirstSyncByte;
 
                             currentPosition = 0;
                             dataLength = packetType = -1;
                         } else {
-                            dataBuffer[currentPosition++] = _byte;
+                            dataBuffer[currentPosition++] = byteBuffer;
                         }
                         break;
                 }
             }
         } catch (
 
         IOException ioexception) {
-            errorListener.ErrorOccured(ioexception);
+            logger.trace("Unable to process message", ioexception);
```

Comment:
Because the `localListener.errorOccured(ioexception);` sets the thing offline with an error description for the end user. I did not follow all possible call hierarchies to check if any IOException is throw, they might be there, if so this log call would help troubleshoot.